### ISSUE TITLE
Correctly handle messages to the HTTP API that are replies to messages with no conversation key.

### ISSUE THREAD

**Issue by hodgestar** [praekeltfoundation/vumi-go#760]
The error looks like:

```
          File "/var/praekelt/vumi-go/go/apps/http_api/resource.py", line 176, in handle_PUT_in_reply_to
            if reply_to_mdh.get_conversation_key() != conversation.key:
          File "/var/praekelt/vumi-go/go/vumitools/utils.py", line 65, in get_conversation_key
            return self._go_metadata['conversation_key']
        exceptions.KeyError: 'conversation_key'
```

**Comment by smn** [comment]
Interesting. I wonder what triggered this from the 3rd party.

**Comment by smn** [comment]
@hodgestar wasn't this fixed as part of #762?

**Comment by hodgestar** [comment]
It was.
